quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,399 @@
+            return ConnectionStringBuilder.createWithAadApplicationCertificate(endpointUrl, appId, x509Certificate, privateKey, tenantId);
+        } catch (IOException | GeneralSecurityException e) {
+            die("Couldn't create ConnectionStringBuilder for app certificate authentication", e);
+            return null;
+        }
+    }
+
+    private static boolean execute(Client kustoClient, String databaseName, String query, ExecutionType executionType) {
+        ClientRequestProperties clientRequestProperties = createClientRequestProperties("SampleApp_" + executionType);
+        KustoOperationResult result;
+        try {
+            result = kustoClient.execute(databaseName, query, clientRequestProperties);
+            System.out.printf("Response from executed %s '%s':%n", executionType.getName(), query);
+            KustoResultSetTable primaryResults = result.getPrimaryResults();
+            for (int rowNum = 1; primaryResults.next(); rowNum++) {
+                KustoResultColumn[] columns = primaryResults.getColumns();
+                List<Object> currentRow = primaryResults.getCurrentRow();
+                System.out.printf("Record %s%n", rowNum);
+                for (int j = 0; j < currentRow.size(); j++) {
+                    Object cell = currentRow.get(j);
+                    System.out.printf("Column: '%s' of type '%s', Value: '%s'%n", columns[j].getColumnName(), columns[j].getColumnType(), cell == null ? "[null]" : cell);
+                }
+                System.out.println();
+            }
+            return true;
+        } catch (DataServiceException e) {
+            System.out.printf("Server error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        } catch (DataClientException e) {
+            System.out.printf("Client error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        } catch (Exception e) {
+            System.out.printf("Unexpected error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope) {
+        return createClientRequestProperties(scope, null);
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope, String timeout) {
+        ClientRequestProperties clientRequestProperties = new ClientRequestProperties();
+        clientRequestProperties.setClientRequestId(String.format("%s;%s", scope, UUID.randomUUID()));
+
+        // Tip: While not common, you can alter the request default command timeout using the below command, e.g. to set the timeout to 10 minutes, use "10m"
+        if (StringUtils.isNotBlank(timeout)) {

```

Comment:
Why is `timeout` is a 4th distinct type of config, in addition to config file (`kusto-sample-config.json`), environment variables, and static constants meant to be configured at the top of this class?

I suspect the answer is that we just want to show **how** `timeout` would be used, but it wouldn't likely be used, and this approach to configs is what @yogilad considers best aligned with that. But I have to ask in case it's a mistake.

Reply:
Strictly for demonstrating how to set it